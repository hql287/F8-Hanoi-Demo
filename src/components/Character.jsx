// React Libraries
import path from 'path';
import React, {Component} from 'react';

// Svg paths
const CharacterPath = path.resolve(__dirname, './static/imgs/Mario.svg');
const SpeechPath = path.resolve(__dirname, './static/imgs/speech-bubble.gif');

import styled from 'styled-components';

export const Layout = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: flex-end;
  ${props => `
    margin-left: ${props.pos.x}px;
  `};
`;

export const Speech = styled.img`
  margin-right: -300px;
`;

export const Mario = styled.img`
  display: block;
  ${props =>
    props.size &&
    `
    max-width: ${props.size}px;
  `};
  ${props =>
    props.isJumping
      ? `
      margin-bottom: ${props.groundHeight + 100}px;
    `
      : `
      margin-bottom: ${props.groundHeight}px;
    `};
`;

class Character extends Component {
  render() {
    const {pos} = this.props;
    return (
      <Layout {...this.props}>
        <Speech {...this.props} src={SpeechPath} alt="" />
        <Mario
          {...this.props}
          size={60}
          src={CharacterPath}
          alt="Mario Charactor"
        />
      </Layout>
    );
  }
}

export default Character;
