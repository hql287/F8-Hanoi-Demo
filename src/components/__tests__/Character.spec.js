// Libs
import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

// Component to test
import Character, {Layout, Mario, Speech} from '../Character';

const groundHeight = 160;
const pos = {x: 0, y: 0};

describe('Renders the Character correctly to the DOM', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Character
        pos={pos}
        groundHeight={groundHeight}
      />
    );
  });

  it('should render the Layout component', () => {
    expect(wrapper.find(Layout)).toHaveLength(1);
    expect(wrapper.find(Layout)).not.toHaveLength(2);
  });

  it('should render Mario', () => {
    expect(wrapper.find(Mario)).toHaveLength(1);
    expect(wrapper.find(Mario)).not.toHaveLength(2);
  });

  it('should render the Speech bubble', () => {
    expect(wrapper.find(Speech)).toHaveLength(1);
    expect(wrapper.find(Speech)).not.toHaveLength(2);
  });

  // Test style Rules
  it('should render components with correct style', () => {

  });

  // Snapshot Testing
  it('matches snapshot', () => {
    const tree = renderer
      .create(
        <Character
          pos={pos}
          groundHeight={groundHeight}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
