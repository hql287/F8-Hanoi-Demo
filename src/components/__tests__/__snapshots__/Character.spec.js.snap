// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the Character correctly to the DOM matches snapshot 1`] = `
<div
  className="Character__Layout-bdFJdW lbsWfW"
>
  <img
    alt=""
    className="Character__Speech-eSrpZz eBgjjc"
    src="/Users/quochungle/Dropbox/Sites/Learning/node/YDKJS/mario/src/components/static/imgs/speech-bubble.gif"
  />
  <img
    alt="Mario Charactor"
    className="Character__Mario-kSBfOl bFPqMa"
    size={60}
    src="/Users/quochungle/Dropbox/Sites/Learning/node/YDKJS/mario/src/components/static/imgs/Mario.svg"
  />
</div>
`;
